src/tool/color.ts

```diff
@@ -180,25 +180,28 @@ export function parse(colorStr: string, rgbaArr?: number[]): number[] {
         return rgbaArr;
     }
 
-    // #abc and #abc123 syntax.
+    // supports the forms #rgb, #rrggbb, #rgba, #rrggbbaa
+    // #rrggbbaa(use the last pair of digits as alpha)
+    // see https://drafts.csswg.org/css-color/#hex-notation
     if (str.charAt(0) === '#') {
-        if (str.length === 4) {
-            let iv = parseInt(str.substr(1), 16);  // TODO(deanm): Stricter parsing.
+        if (str.length === 4 || str.length === 5) {
+            const iv = parseInt(str.slice(1, 4), 16);  // TODO(deanm): Stricter parsing.
```

Comment:
`str.length` can be cached in a variable since they are used more frequently now

Reply:
Tweaked in the commit a74bc35a2dea5966c81cf3c5a9e51359619ef662